src/ripple/shamap/impl/SHAMap.cpp

```diff
@@ -419,28 +423,25 @@ SHAMap::unshareNode (std::shared_ptr<Node> node, SHAMapNodeID const& nodeID)
 SHAMapTreeNode*
 SHAMap::firstBelow(SHAMapAbstractNode* node, NodeStack& stack) const
 {
-    // Return the first item below this node
-    while (true)
+    // Return the first item at or below this node
+    assert(node != nullptr);
```

Comment:
Not in this change set, but I prefer non-null parameters be passed by reference. Maybe for another PR.